### PR TITLE
gh-123123: Fix display of syntax errors covering multiple lines

### LINKED ISSUE #123123: Ambiguous invalid syntax error for a missing comma
# Bug report

### Bug description:

Trying to run the following code:

```python
dummy_call(
    "dummy value"
    foo="bar",
)
```

results in a `SyntaxError` that looks like:

```
  File "...", line 2
    "dummy value"
    ^^^
SyntaxError: invalid syntax. Perhaps you forgot a comma?
```

On 3.13+, the first three characters are also highlighted in red. I would expect the end of the line (where the comma should be) to be marked, the `foo=bar` line, or possibly the entire line.

Interestingly, if you remove the indentation, then the error looks as expected:

```
  File "...", line 1
    dummy_call("dummy value" foo="bar",)
               ^^^^^^^^^^^^^^^^^
SyntaxError: invalid syntax. Perhaps you forgot a comma?
```

### CPython versions tested on:

3.12, 3.13, CPython main branch

### Operating systems tested on:

Linux

### Linked PRs
* gh-123131
* gh-123147

### PR DESCRIPTION
**Pull request by pablogsal**
Closes: #123123

* Issue: gh-123123

### CODE REVIEW

File: Lib/traceback.py [hunk 1]
```diff
@@ -1292,7 +1292,10 @@ def _format_syntax_error(self, stype, **kwargs):
                 yield '    {}\n'.format(ltext)
             else:
                 offset = self.offset
-                end_offset = self.end_offset if self.end_offset not in {None, 0} else offset
+                if self.lineno == self.end_lineno:
+                    end_offset = self.end_offset if self.end_offset not in {None, 0} else offset
+                else:
+                    end_offset = len(self.text)
```
Comment (skirpichev) [inline Lib/traceback.py]:
```suggestion
                    end_offset = len(self.text) + 1
```
?

Without this, issue's traceback looks like
```pycon
>>> dummy_call(
...     "dummy value"
...     foo="bar",
... )
... 
  File "<unknown>", line 2
    "dummy value"
    ^^^^^^^^^^^^
SyntaxError: invalid syntax. Perhaps you forgot a comma?
```
I think all line should be marked, till the end (where comma shoule be).

**Comment by skirpichev** [comment]
It seems, test_syntax_error_various_offsets() should be adjusted.

File: Lib/traceback.py (hunk 1 above)
Comment (pablogsal) [reply Lib/traceback.py]: I consider it but it looks weird to highlight white space. Notice this path will be taken for multi line syntax errors so it's not just about missing commas.

Comment (skirpichev) [reply Lib/traceback.py]: In this particular case it's not a white space.  Also, your solution doesn't avoid such scenario.  Just add two white spaces:)

Probably, that part - another issue: trailing spaces shouldn't be included in exception's metadata.

Comment (pablogsal) [reply Lib/traceback.py]: Oh apologies, I think I misread your comment! You are absolutely right. Let me amend this in a slightly different way to merge it with a code path that's after this

Comment (pablogsal) [reply Lib/traceback.py]: Hummmm, this is what I get when testing in my machine with the current code:

```
./python lel.py
  File "/home/pablogsal/github/python/main/lel.py", line 2
    "dummy value"
    ^^^^^^^^^^^^^
SyntaxError: invalid syntax. Perhaps you forgot a comma?
```

and this is with your suggestion:

```
./python lel.py
  File "/home/pablogsal/github/python/main/lel.py", line 2
    "dummy value"
    ^^^^^^^^^^^^^^
SyntaxError: invalid syntax. Perhaps you forgot a comma?
```

Comment (pablogsal) [reply Lib/traceback.py]: Am I missing something?

Comment (pablogsal) [reply Lib/traceback.py]: Ah, seems that when executing over a file there is an extra newline character in the text metadata!

**Review by tuuputamadre — APPROVED** [review]

**Pull request merged**